starknet-rpc-test/estimate_message_fee.rs

```diff
@@ -0,0 +1,72 @@
+extern crate starknet_rpc_test;
+
+use assert_matches::assert_matches;
+use rstest::rstest;
+use starknet_core::types::{BlockId, BlockTag, EthAddress, MsgFromL1, StarknetError};
+use starknet_core::utils::get_selector_from_name;
+use starknet_ff::FieldElement;
+use starknet_providers::ProviderError::StarknetError as StarknetProviderError;
+use starknet_providers::{MaybeUnknownErrorCode, Provider, StarknetErrorWithMessage};
+use starknet_rpc_test::constants::{L1_CONTRACT_ADDRESS, TEST_CONTRACT_ADDRESS};
+use starknet_rpc_test::fixtures::{madara, ThreadSafeMadaraClient};
+
+#[rstest]
+#[tokio::test]
+async fn fail_non_existing_block(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
+
+    let message: MsgFromL1 = MsgFromL1 {
+        from_address: EthAddress::from_hex(L1_CONTRACT_ADDRESS).unwrap(),
+        to_address: FieldElement::from_hex_be(TEST_CONTRACT_ADDRESS).unwrap(),
+        entry_point_selector: get_selector_from_name("sqrt").unwrap(),
+        payload: vec![FieldElement::from_hex_be("0x0").unwrap()],
+    };
+
+    assert_matches!(
+        rpc.estimate_message_fee(message, BlockId::Hash(FieldElement::ZERO)).await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code: MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::BlockNotFound
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn fail_if_message_fail(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
+
+    let message: MsgFromL1 = MsgFromL1 {
+        from_address: EthAddress::from_hex(L1_CONTRACT_ADDRESS).unwrap(),
+        to_address: FieldElement::from_hex_be(TEST_CONTRACT_ADDRESS).unwrap(),
+        entry_point_selector: get_selector_from_name("sqrt").unwrap(),
+        payload: vec![FieldElement::from_hex_be("0x0").unwrap()],
+    };
+
+    assert_matches!(
+        rpc.estimate_message_fee(message, BlockId::Tag(BlockTag::Latest)).await,
+        Err(StarknetProviderError(StarknetErrorWithMessage { code: MaybeUnknownErrorCode::Known(code), .. })) if code == StarknetError::ContractError
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn works_ok(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
+
+    let message: MsgFromL1 = MsgFromL1 {
+        from_address: EthAddress::from_hex(L1_CONTRACT_ADDRESS).unwrap(),
+        to_address: 3u64.into(),
+        entry_point_selector: 2u64.into(),
+        payload: vec![1u64.into()],
+    };
```

Comment:
Bha la par contre il va failloir utiliser des valeurs qui fonctionnent.

La fonction que tu veux call c'est celle là:
```rust
@l1_handler
func test_l1_handler_store_under_caller_address{syscall_ptr: felt*}(from_address: felt, value: felt) {
    storage_write(address=from_address, value=value);
    return ();
}
```
dans  cairo-contracts/src/test_contracts/test.cairo

Donc là il faut que tu changes to entry_point_selector:
```shell
$ starkli selector test_l1_handler_store_under_caller_address
0x014093c40d95d0a3641c087f7d48d55160e1a58bc7c07b0d2323efeeb3087269
```

Et ton addresse:
```rust
pub const TEST_CONTRACT_ADDRESS: &str = "0x0000000000000000000000000000000000000000000000000000000000001111";
```
dans `starknet-rpc-test/src/constants.rs`

Sinon il appelle un entrypoint qui existe pas sur un contract qui existe pas.